autoPyTorch/pipeline/components/training/trainer/cutout_utils.py

```diff
@@ -0,0 +1,144 @@
+                              ) -> Callable:
+        return lambda criterion, pred: lam * criterion(pred, y_a) + (1 - lam) * criterion(pred, y_b)
+
+    @staticmethod
+    def get_hyperparameter_search_space(
+            dataset_properties: Optional[Dict] = None,
+            weighted_loss: HyperparameterSearchSpace = HyperparameterSearchSpace(
+                hyperparameter="weighted_loss",
+                value_range=[True, False],
+                default_value=True),
+            la_steps: HyperparameterSearchSpace = HyperparameterSearchSpace(
+                hyperparameter="la_steps",
+                value_range=(5, 10),
+                default_value=6,
+                log=False),
+            la_alpha: HyperparameterSearchSpace = HyperparameterSearchSpace(
+                hyperparameter="la_alpha",
+                value_range=(0.5, 0.8),
+                default_value=0.6,
+                log=False),
+            use_lookahead_optimizer: HyperparameterSearchSpace = HyperparameterSearchSpace(
+                hyperparameter="use_lookahead_optimizer",
+                value_range=(True, False),
+                default_value=True),
+            use_stochastic_weight_averaging: HyperparameterSearchSpace = HyperparameterSearchSpace(
+                hyperparameter="use_stochastic_weight_averaging",
+                value_range=(True, False),
+                default_value=True),
+            use_snapshot_ensemble: HyperparameterSearchSpace = HyperparameterSearchSpace(
+                hyperparameter="use_snapshot_ensemble",
+                value_range=(True, False),
+                default_value=True),
+            se_lastk: HyperparameterSearchSpace = HyperparameterSearchSpace(
+                hyperparameter="se_lastk",
+                value_range=(3,),
+                default_value=3),
+            patch_ratio: HyperparameterSearchSpace = HyperparameterSearchSpace(
+                hyperparameter="patch_ratio",
+                value_range=(0, 1),
+                default_value=0.2),
+            cutout_prob: HyperparameterSearchSpace = HyperparameterSearchSpace(
+                hyperparameter="cutout_prob",
+                value_range=(0, 1),
+                default_value=0.2),
+    ) -> ConfigurationSpace:
+
+        cs = ConfigurationSpace()
+
+        add_hyperparameter(cs, patch_ratio, UniformFloatHyperparameter)
+        add_hyperparameter(cs, cutout_prob, UniformFloatHyperparameter)
+        add_hyperparameter(cs, use_stochastic_weight_averaging, CategoricalHyperparameter)
+        snapshot_ensemble_flag = False
+        if any(use_snapshot_ensemble.value_range):
+            snapshot_ensemble_flag = True
+
+        use_snapshot_ensemble = get_hyperparameter(use_snapshot_ensemble, CategoricalHyperparameter)
+        cs.add_hyperparameter(use_snapshot_ensemble)
+
+        if snapshot_ensemble_flag:
+            se_lastk = get_hyperparameter(se_lastk, Constant)
```

Comment:
True, we should fix this

Reply:
It is constant in our case since that is what we did for the publication, however, it can be a hyperparameter. Only thing is it has to be a hyperparameter that is lower or equal to the number of restarts, it needs this condition.